presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+            table.getSchemaName(),
+            table.getTableName(),
+            SchemaParser.toJson(icebergTable.schema()),
+            PartitionSpecParser.toJson(icebergTable.spec()),
+            getColumns(icebergTable.schema(), typeManager),
+            getDataPath(icebergTable.location()),
+            getFileFormat(icebergTable));
+    }
+
+    @Override
+    public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(ConnectorSession session,
+            ConnectorInsertTableHandle insertHandle,
+            Collection<Slice> fragments,
+            Collection<ComputedStatistics> computedStatistics,
+            List<ConnectorTableHandle> sourceTableHandles)
+    {
+        IcebergWritableTableHandle table = (IcebergWritableTableHandle) insertHandle;
+        org.apache.iceberg.Table icebergTable = transaction.table();
+        List<CommitTaskData> commitTasks = new ArrayList<>();
+
+        if (!fragments.isEmpty()) {
+            commitTasks = fragments.stream()
+                .map(slice -> commitTaskCodec.fromJson(slice.getBytes()))
+                .collect(toImmutableList());
+
+            Type[] partitionColumnTypes = icebergTable.spec().fields().stream()
+                .map(field -> field.transform().getResultType(
+                    icebergTable.schema().findType(field.sourceId())))
+                .toArray(Type[]::new);
+
+            AppendFiles appendFiles = transaction.newFastAppend();
+            for (CommitTaskData task : commitTasks) {
+                DataFiles.Builder builder = DataFiles.builder(icebergTable.spec())
+                        .withInputFile(HadoopInputFile.fromLocation(task.getPath(), getConfiguration(session, table.getSchemaName())))
+                        .withFormat(table.getFileFormat())
+                        .withMetrics(task.getMetrics().metrics());
+
+                if (!icebergTable.spec().fields().isEmpty()) {
+                    String partitionDataJson = task.getPartitionDataJson()
+                            .orElseThrow(() -> new VerifyException("No partition data for partitioned table"));
+                    builder.withPartition(PartitionData.fromJson(partitionDataJson, partitionColumnTypes));
+                }
+
+                appendFiles.appendFile(builder.build());
+            }
+
+            // Update the map that tracks tables on which the materialized view depends and the corresponding snapshot ids of the table
+            String tableSnapshotIDString = Joiner.on(",").withKeyValueSeparator("=").join(transactionHandle.getTableSnapshotIdMap());
+            appendFiles.set(DEPENDS_ON_TABLES, tableSnapshotIDString);
+            appendFiles.commit();
+        }
+        transaction.commitTransaction();
+        return Optional.of(new HiveWrittenPartitions(commitTasks.stream()
+            .map(CommitTaskData::getPath)
+            .collect(toImmutableList())));
+    }
+
+    private boolean isAMaterializedView(Table table)
```

Comment:
Drop the "A" from the method name to match the common naming convention

Reply:
done